kcidb/__init__.py

```diff
@@ -234,3 +237,62 @@ def validate_main():
         print(err, sys.stderr)
         return 2
     return 0
+
+
+def tests_list_main():
+    """Execute the kcidb-tests-list command-line tool"""
+    description = 'kcidb-tests-list - list known test programs'
+    parser = argparse.ArgumentParser(description=description)
+    parser.parse_args()
+
+    catalog = yaml.safe_load(sys.stdin)
+    tests_schema.validate(catalog)
+
+    def print_test(path, test):
+        if 'home' in test:
+            print(".".join(path), test['title'])
+        else:
+            for name, subtest in test['tests'].items():
+                print_test(path + [name], subtest)
+    print_test([], catalog)
+
+    return 0
+
+
+def tests_validate_main():
+    """Execute the kcidb-tests-validate command-line tool"""
+    description = 'kcidb-tests-validate - Validate test catalog YAML'
+    parser = argparse.ArgumentParser(description=description)
+    parser.add_argument(
+        "-u", "--urls",
+        action='store_true',
+        help="Verify URLs in the catalog are accessible"
+    )
+    args = parser.parse_args()
+
+    try:
+        catalog = yaml.safe_load(sys.stdin)
+    except yaml.YAMLError as err:
+        print(err, sys.stderr)
+        return 1
+
+    try:
+        tests_schema.validate(catalog)
+    except jsonschema.exceptions.ValidationError as err:
+        print(err, sys.stderr)
+        return 2
```

Comment:
Is there a convention about returning anything else than 0 or 1 in this context?

Reply:
Not really, but nothing usually checks for anything else but non-zero, and in theory it could be useful :)

Reply:
That's a bit far-fetched but sure why not :)